Timestamp parse failure

olessia opened the pull request:
Ticket: https://github.com/Shopify/camus-project/issues/13

Pass in an extra flag to the record wrapper that indicates whether the timestamp was parsed correctly. 
Count how many times this happens and publish the metric to statsd.

Commit:
Report the case when the timestamp was not found/parsed.

dterror commented on camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java:
```diff
@@ -277,6 +286,11 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
+
+        Class klass = com.linkedin.camus.etl.kafka.common.KafkaMessage.class;
+        String fullPath = klass.getResource('/' + klass.getName().replace('.', '/') + ".class").getPath();
+        log.error("DOOTDOOT this is the klass path: " + fullPath);
```
This is from Dave's commit, we can probably remove it

dterror commented on camus-kafka-coders/src/main/java/com/linkedin/camus/etl/kafka/coders/JsonStringMessageDecoder.java:
```diff
@@ -120,15 +120,18 @@ else if (timestampFormat.equals("ISO-8601")) {
       }
     }
+    else {
+      log.warn("Message does not have the field " + timestampField); // TODO
```
I like it, another datadog gauge?

olessia replied:
We could, but then I'd need to store another flag in the wrapper. For now I just wanted to validate that when the warning was printed it was because of this.

dterror replied:
gotcha

dterror-zz reviewed (approved):
👍

drdee commented:
Won't this lead to excessive logging?

> On Oct 26, 2017, at 09:32, Diogo Terror <notifications@github.com> wrote:
> 
> @dterror commented on this pull request.
> 
> In camus-kafka-coders/src/main/java/com/linkedin/camus/etl/kafka/coders/JsonStringMessageDecoder.java:
> 
> > @@ -120,15 +120,18 @@ else if (timestampFormat.equals("ISO-8601")) {
>          }
>        }
>      }
> +    else {
> +      log.warn("Message does not have the field " + timestampField); // TODO
> gotcha
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

dterror commented:
not much worse than we have now and it's for us to better understand it so we can remove the excessive logging we currently have.

drdee commented:
👍

Commit:
Submit more data to statsd:
job start
run-time stats
pull-max-time-reached: submit 0 when the partition was fully pulled (this should help with alerting)
